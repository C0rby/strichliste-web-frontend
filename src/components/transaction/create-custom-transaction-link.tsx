import * as React from 'react';
import { RouteComponentProps, withRouter } from 'react-router';
import { Button, theme } from '../ui';

export interface CreateCustomTransactionLinkProps
  extends RouteComponentProps<{}> {
  isDeposit: boolean;
}

function Link({
  isDeposit,
  history,
  match,
}: CreateCustomTransactionLinkProps): JSX.Element {
  return (
    <Button
      onClick={() => {
        history.push(match.url + (isDeposit ? '/deposit' : '/dispense'));
      }}
      color={isDeposit ? theme.green : theme.red}
    >
      ?
    </Button>
  );
}

export const CreateCustomTransactionLink = withRouter(Link);
