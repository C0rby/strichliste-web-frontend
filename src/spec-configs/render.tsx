import { ThemeProvider, theme } from 'bricks-of-sand';
import { MemoryHistory, createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { RenderResult, render } from 'react-testing-library';
import { DeepPartial, Store, createStore } from 'redux';
import { StoreContext } from 'redux-react-hook';

import { AppState, reducer } from '../store';

const themeConfig = { light: theme, dark: theme };

export function renderWithContext(
  ui: JSX.Element,
  initialState: DeepPartial<AppState>,
  store: Store<AppState> = createStore(reducer, initialState),
  _route: string = '/',
  history: MemoryHistory = createMemoryHistory()
): RenderResult {
  return render(
    <ThemeProvider themes={themeConfig}>
      <StoreContext.Provider value={store}>
        <Provider store={store}>
          <Router history={history}>
            <IntlProvider>{ui}</IntlProvider>
          </Router>
        </Provider>
      </StoreContext.Provider>
    </ThemeProvider>
  );
}

export function renderAndReturnContext(
  ui: JSX.Element,
  initialState: DeepPartial<AppState>,
  store: Store<AppState> = createStore(reducer, initialState),
  _route: string = '/',
  history: MemoryHistory = createMemoryHistory()
): { result: RenderResult; store: Store<AppState>; history: MemoryHistory } {
  return {
    result: render(
      <ThemeProvider themes={themeConfig}>
        <Provider store={store}>
          <Router history={history}>
            <IntlProvider textComponent={React.Fragment}>{ui}</IntlProvider>
          </Router>
        </Provider>
      </ThemeProvider>
    ),
    store,
    history,
  };
}

export function renderWithIntl(ui: JSX.Element): RenderResult {
  return render(
    <ThemeProvider themes={themeConfig}>
      <IntlProvider textComponent={React.Fragment}>{ui}</IntlProvider>
    </ThemeProvider>
  );
}
