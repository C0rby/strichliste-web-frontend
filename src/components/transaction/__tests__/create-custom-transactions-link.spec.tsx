import * as React from 'react';
import { cleanup, fireEvent } from 'react-testing-library';

import { renderAndReturnContext } from '../../../spec-configs/render';
import { CreateCustomTransactionLink } from '../create-custom-transaction-link';

afterEach(cleanup);

describe('CreateCustomTransactionLink', () => {
  describe('with deposit', () => {
    it('matches the snapshot', () => {
      const { result, history } = renderAndReturnContext(
        <CreateCustomTransactionLink isDeposit />,
        {}
      );

      const button = result.getByText('?');
      fireEvent.click(button);

      expect(history.location.pathname).toBe('//deposit');
      expect(result.container.firstChild).toMatchSnapshot();
    });
  });
  describe('with dispense', () => {
    it('matches the snapshot', () => {
      const { result, history } = renderAndReturnContext(
        <CreateCustomTransactionLink isDeposit={false} />,
        {}
      );

      const button = result.getByText('?');
      fireEvent.click(button);

      expect(history.location.pathname).toBe('//dispense');
      expect(result.container.firstChild).toMatchSnapshot();
    });
  });
});
