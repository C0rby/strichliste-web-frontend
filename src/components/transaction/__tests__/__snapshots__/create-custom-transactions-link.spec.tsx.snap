// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCustomTransactionLink with deposit matches the snapshot 1`] = `
<div
  class="css-mglgiq"
>
  <button
    class="css-1bzihpp"
    color="#9DB06A"
    type="button"
  >
    ?
  </button>
</div>
`;

exports[`CreateCustomTransactionLink with dispense matches the snapshot 1`] = `
<div
  class="css-mglgiq"
>
  <button
    class="css-1fvk27e"
    color="#E86C76"
    type="button"
  >
    ?
  </button>
</div>
`;
